src/initialize_atmos.c

```diff
@@ -1586,6 +1590,178 @@ void initialize_atmos(atmos_data_struct        *atmos,
+    for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+      for (j = 0; j < NF; j++) {
+        veg_hist[rec][v].crop_frac[j] = veg_lib[veg_con[v].veg_class].crop_frac[dmy[rec].month-1];
+      }
+    }
+  }
+
+  if(param_set.TYPE[CROP_FRAC].SUPPLIED) {
+    if(param_set.FORCE_DT[param_set.TYPE[CROP_FRAC].SUPPLIED-1] == 24) {
+      /* daily crop_frac provided */
+      for (rec = 0; rec < global_param.nrecs; rec++) {
+        for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+          if (veg_con[v].crop_frac_active) {
+            sum = 0;
+            for (j = 0; j < NF; j++) {
+              hour = rec*global_param.dt + j*options.SNOW_STEP + global_param.starthour - hour_offset_int;
+              if (global_param.starthour - hour_offset_int < 0) hour += 24;
+              idx = (int)((float)hour/24.0);
+              if (local_forcing_data[CROP_FRAC][idx] != NODATA_VH) {
+	        veg_hist[rec][v].crop_frac[j] = local_forcing_data[CROP_FRAC][idx]; // assume constant over the day
+                if (veg_hist[rec][v].crop_frac[j] < MIN_CROP_FRAC) veg_hist[rec][v].crop_frac[j] = MIN_CROP_FRAC;
+              }
+              sum += veg_hist[rec][v].crop_frac[j];
+            }
+            if(NF>1) veg_hist[rec][v].crop_frac[NR] = sum / (float)NF;
+          }
+        }
+      }
+    }
+    else {
+      /* sub-daily crop_frac provided */
+      for(rec = 0; rec < global_param.nrecs; rec++) {
+        for(v = 0; v < param_set.TYPE[CROP_FRAC].N_ELEM; v++) {
+          if (veg_con[v].crop_frac_active) {
+            sum = 0;
+            for(i = 0; i < NF; i++) {
+              hour = rec*global_param.dt + i*options.SNOW_STEP + global_param.starthour - hour_offset_int;
+              veg_hist[rec][v].crop_frac[i] = 0;
+              while (hour < rec*global_param.dt + (i+1)*options.SNOW_STEP + global_param.starthour - hour_offset_int) {
+                idx = hour;
+                if (idx < 0) idx += 24;
+                if (local_forcing_data[CROP_FRAC][idx] != NODATA_VH) {
+	          veg_hist[rec][v].crop_frac[i] = local_forcing_data[CROP_FRAC][idx];
+                  if (veg_hist[rec][v].crop_frac[i] < MIN_CROP_FRAC) veg_hist[rec][v].crop_frac[i] = MIN_CROP_FRAC;
+                }
+                hour++;
+              }
+	      sum += veg_hist[rec][v].crop_frac[i];
+            }
+            if(NF>1) veg_hist[rec][v].crop_frac[NR] = sum / (float)NF;
+          }
+        }
+      }
+    }
+  }
+
+  /* HACK: if crop, and crop frac is active, normalize vegcover by crop_frac */
+  for (rec = 0; rec < global_param.nrecs; rec++) {
+    for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+      if (options.CROPFRAC && veg_con[v].crop_frac_active) {
+        if (veg_hist[rec][v].crop_frac[0] > 0) {
+          for (j = 0; j < NF; j++) {
+            veg_hist[rec][v].vegcover[j] /= veg_hist[rec][v].crop_frac[0];
+	    veg_hist[rec][v].vegcover[j] =1; //ingjerd added. keep vegcover at 1 for crops
```

Comment:
Actually, looking back at full_energy.c, I see that on line 309, you are doing the vegcover /= crop_frac step.  So, really, line 1661 here is incorrect (else line 309 of full_energy.c becomes the second division by crop_frac, which is wrong).  So, I definitely recomend deleting line 1661 and changing the comment on line 1655 to be more descriptive of what you're actually doing - setting vegcover to 1 for crops.